tools/github-actions/cascading/src/helpers.ts

```diff
@@ -0,0 +1,81 @@
+import {BaseLogger} from './cascading';
+
+/**
+ * Ensure that the sorting based on git have been done correctly, update the list if there is an issue
+ *
+ * @param branches
+ */
+export function verifyBranches(branches: string[]) {
+  return branches.sort((branchA, branchB) => {
+    const aExtract = branchA.match(/release\/([0-9]+)\.([0-9]+)/);
```

Comment:
this implementation looks inefficient.

a sort implementation is usually O(n*log(n)) which means you will check the same branch name multiple times.
you could extract the version information used to determine the natural order in a separate variable in a `.map()`